src/synapse/api.py

```diff
@@ -73,16 +92,21 @@ def _generate_mac(
     nonce: str,
     user: str,
     password: str,
-    admin: bool = False,
+    admin: str | bool = False,
     user_type: typing.Optional[str] = None,
 ) -> str:
     """Generate mac to register user.
 
+    "The MAC is the hex digest output of the HMAC-SHA1 algorithm, with the key being the shared
```

Comment:
are we sure we should use `SHA1`? Its quite short

Reply:
Yes, I didnt find any discussion regarding that but here is the [code](https://github.com/matrix-org/synapse/blob/develop/synapse/rest/admin/users.py#L587).